src/charm.py

```diff
@@ -238,24 +209,16 @@ def __init__(self, *args):
             ),
         )
 
-        self.service_patch = KubernetesServicePatch(
-            charm=self,
-            service_type="LoadBalancer",
+        self.lb_patch = KubernetesLoadBalancer(
+            name=f"{self.app.name}-lb",
+            namespace=self.model.name,
+            field_manager=self.app.name,
             ports=self._service_ports,
             additional_annotations=self._loadbalancer_annotations,
-            refresh_event=[
-                ipa_v1.on.data_provided,  # type: ignore
-                ipa_v2.on.data_provided,  # type: ignore
-                ipa_v1.on.data_removed,  # type: ignore
-                ipa_v2.on.data_removed,  # type: ignore
-                self.ingress_per_unit.on.data_provided,  # type: ignore
-                self.ingress_per_unit.on.data_removed,  # type: ignore
-                self.traefik_route.on.ready,  # type: ignore
-                self.traefik_route.on.data_removed,  # type: ignore
-                self.on.traefik_pebble_ready,  # type: ignore
-                self.on.config_changed,  # type: ignore
-            ],
+            additional_labels={"app.kubernetes.io/name": self.app.name},
+            additional_selectors={"app.kubernetes.io/name": self.app.name},
```

Comment:
Now that it isn't a charm library that needs to observe its own charm events, should this be in `__init__`?  I bet it fits more naturally somewhere else

Putting this somewhere else will also make a lot of the mocking in tests unnecessary as we no longer make lightkube calls on any charm invocation